python/fish.py

```diff
@@ -1325,62 +1156,48 @@ def fish_msg_wo_marker(msg):
-        "Add, change or remove key for target or perform DH1080\n"
-        "keyexchange with <nick>.\n"
-        "Target can be a channel or a nick.\n"
-        "\n"
-        "Without arguments this command lists all keys.\n"
-        "\n"
-        "Examples:\n"
-        "Set the key for a channel: /blowkey set -server freenet #blowfish key\n"
-        "Remove the key:            /blowkey remove #blowfish\n"
-        "Set the key for a query:   /blowkey set nick secret+key\n"
-        "List all keys:             /blowkey\n\n"
-        "\n** stores keys in plaintext by default **\n\n"
-        "DH1080:                    /blowkey exchange nick\n"
-        "\nPlease read the source for a note about DH1080 key exchange\n",
-        "list"
-        "|| genkey"
-        "|| set %(irc_channel)|%(nicks)|-server %(irc_servers) %- "
-        "|| remove %(irc_channel)|%(nicks)|-server %(irc_servers) %- "
-        "|| exchange %(nick)|-server %(irc_servers) %-",
-        "fish_cmd_blowkey",
-        "",
-    )
+            "blowkey", "Manage FiSH keys",
+            "[list] | set [-server <server>] [<target>] <key> "
+            "| remove [-server <server>] <target> "
+            "| exchange [-server <server>] [<nick>] "
+            "| genkey",
+            "Add, change or remove key for target or perform DH1080 key"
+            "exchange with <nick>.\n"
+            "Target can be a channel or a nick.\n"
+            "\n"
+            "Without arguments this command lists all keys.\n"
+            "\n"
+            "Examples:\n"
+            "Set the key for a channel: /blowkey set -server freenet #blowfish"
+            " key\n"
+            "Remove the key:            /blowkey remove #blowfish\n"
+            "Set the key for a query:   /blowkey set nick secret+key\n"
+            "List all keys:             /blowkey\n"
+            "DH1080:                    /blowkey exchange nick\n"
+            "\nPlease read the source for a note about DH1080 key exchange\n",
+            "list || set %(irc_channel)|%(nicks)|-server %(irc_servers) %- "
+            "|| remove %(irc_channel)|%(nicks)|-server %(irc_servers) %- "
+            "|| exchange %(nick)|-server %(irc_servers) %-"
+            "|| genkey",
+            "fish_cmd_blowkey", "")
 
     fish_config_init()
     fish_config_read()
     fish_secure()
 
-    weechat.hook_modifier("irc_in2_notice", "fish_modifier_in_notice_cb", "")
-    weechat.hook_modifier("irc_in2_privmsg", "fish_modifier_in_privmsg_cb", "")
-    weechat.hook_modifier("irc_in2_topic", "fish_modifier_in_topic_cb", "")
-    weechat.hook_modifier("irc_in2_332", "fish_modifier_in_332_cb", "")
+    weechat.hook_modifier("irc_in_notice", "fish_modifier_in_notice_cb", "")
+    weechat.hook_modifier("irc_in_privmsg", "fish_modifier_in_privmsg_cb", "")
+    weechat.hook_modifier("irc_in_topic", "fish_modifier_in_topic_cb", "")
+    weechat.hook_modifier("irc_in_332", "fish_modifier_in_332_cb", "")
```

Comment:
Hi,

This change from `irc_in2_*` to `irc_in_*` makes the score of the script of 60/100 with weechat-script-lint.

The modifiers `irc_in2_*` are recommended since WeeChat 2.7, see detail in the doc here: https://weechat.org/files/doc/devel/weechat_scripting.en.html#python_strings

Is there a good reason to use these modifiers `irc_in_*`?

Note: the modifiers were changed in this commit: 1233ce2e2fc2f6fbb69279e5c1a2dcbb9fd6b182

Reply:
Hey.

Yes, we're doing decryption of messages. So the charset can potentially change from input value to output value. Since we don't want to have to guess charset again we work with `irc_in_*` and let weechat handle the decoding to UTF-8 later.